src/main/java/service/operation/SupplyHandler.java

```diff
@@ -0,0 +1,13 @@
+package service.operation;
+
+import db.Storage;
+import models.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        Integer oldQuantity = Storage.fruitMap.get(fruitTransaction.getFruit());

```

Comment:
let's add some validation, e.g. check if the product `quantity <0` and throw an exception if needed.